src/timeline/RoomlistModel.cpp

```diff
@@ -899,6 +899,40 @@ FilteredRoomlistModel::toggleTag(QString roomid, QString tag, bool on)
         }
 }
 
+void
+FilteredRoomlistModel::nextRoomWithActivity()
+{
+        QString targetRoomId = nullptr;
+        auto r               = currentRoom();
+        auto currentRoomId   = r ? r->roomId() : nullptr;
+        // first look for mentions
+        for (const auto &room : roomlistmodel->models) {
```

Comment:
So, this iterates rooms in pretty much random order. A bit cooler would be, if you could iterate them in the order they are displayed (imo). You would need to create model indices for that and then use the `data(idx, RoomlistModel::HasLoudNotification)` to access the interesting fields. Although if you don't want to, that is fine too.

Reply:
That's what I tried initially, but in my experiments `if (sourceModel()->data(index(i, 0), RoomlistModel::HasUnreadMessages).toBool())` always returned true... Then I had a look at how HasUnreadMessages was populated and this is where I decided to use roomReadStatus.

Reply:
I don't see why that should return anything different than what the code you use now. But in any case, you shouldn't call `data()` on the source model, but on the current model (`this->data()`). And I think going by mentions, then by unread message count > 0 makes the most sense.

Reply:
thanks, with `this->` it works. I pushed a new implementation that should do it.